docs/configuring.md

```diff
@@ -0,0 +1,254 @@
+
+- **SHMMAX**: The maximum size of the shared memory segment (bytes).
+- **SHMALL**: The total amount of shared memory in the system (bytes or pages).
+
+The user running the Java process must either be ROOT or have permissions to use huge pages. For the appropriate permissions, the user must be a member of a group that has its group identifier (gid) stored in **/proc/sys/vm/hugetlb_shm_group**. The locked memory limit must also be increased to at least the size of the Java heap by using the **ulimit -l** command.
+
+Where huge page support is available, the following default sizes apply for the object heap:
+
+- Linux on x86: 2 MB
+- Linux on IBM POWER: 16 MB and 2MB (See Note)
+- Linux on IBM Z: 1 MB
+
+<i class="fa fa-pencil-square-o" aria-hidden="true"></i> Note: IBM POWER9 and later systems support either 16 MB or 2 MB huge pages. However PowerVM LPARs support only 16 MB huge pages, while bare-metal machines support only 2 MB huge pages.
+
+
+Transparent HugePage Support (THP) is an automated mechanism of using huge pages to back virtual memory. On Linux kernels that support THP, it is typically enabled by default with the **madvise** option and can be relied on to provide huge pages as required without any user configuration. To disable THP for your application, use the OpenJ9 [`-XX:-TransparentHugePage`](xxtransparenthugepage.md) option on the command line. To disable THP system-wide, change the **sysfs** boot time defaults with the command `transparent_hugepage=never `. For more information about THP see [Transparent HugePage Support](https://www.kernel.org/doc/html/latest/admin-guide/mm/transhuge.html).
+
+### Windows systems
+
+On Windows systems, large pages are typically 2 MB in size. To use large pages, the VM user must have the Windows **Lock pages in memory** setting enabled in the Local Security Policy. Applications must also be run with Admin privileges in order to use large page memory allocations.
+
+For more information, see the following resources from Microsoft:
+
+- [Large page support](https://docs.microsoft.com/en-gb/windows/win32/memory/large-page-support)
+- [`GetLargePageMinimum` function (`memoryapi.h`)](https://docs.microsoft.com/en-gb/windows/win32/api/memoryapi/nf-memoryapi-getlargepageminimum)
+
+### z/OS systems  
+
+When available, 1 MB pageable pages are the default size for the object heap and the code cache. Other page sizes are available for the object heap, depending on the system architecture as shown in the following table:
+
+| Large page size  | System architecture required              | `-Xlp:codecache` | `-Xlp:objectheap`          |
+|------------------|-------------------------------------------|------------------|----------------------------|
+| 2 GB nonpageable | IBM zEnterprise EC12 processor or later   | Not supported    | Supported (64-bit VM only) |
+| 1 MB nonpageable | System z10 processor or later             | Not supported    | Supported (64-bit VM only) |
```

Comment:
While accurate, I believe these docs will only cover 64-bit JVM only, so not sure if the `(64-bit VM only)` is necessary.

Reply:
Yes, Peter suggested we don't bring in any 31-bit z/OS content. However, I've got some cleaning up to do as some cmdline options were already in place. E.g. -XCEEHDLR. I'll sweep the docs in a separate PR to ensure we're only indicating 64-bit for z/OS. We do have some 31-bit content for Linux on Z, so I'll have to be careful not to accidentally remove anything in the process.  :-/ - I'll involve you in the review there if that's ok with you?